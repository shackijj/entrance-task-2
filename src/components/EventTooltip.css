.EventTooltip {
  position: relative;
  display: inline-block;
  background: #FFFFFF;
  padding: 16px;
  box-shadow: 0 1px 16px 0 rgba(0,44,92,0.28);
  border-radius: 8px;
  min-width: 330px;
}

.EventTooltip-Title {
  font-size: 1.154em;
  line-height: 1.846em;
  font-weight: bold;
}

.EventTooltip-Users {
  margin-top: 16px;
  display: inline-flex;
  align-items: center;
}

.EventTooltip-Button {
  position: absolute;
  top: 8px;
  right: 8px;
}

.EventTooltip-MoreUsers {
  color: rgba(0, 0, 0, 0.5);
}