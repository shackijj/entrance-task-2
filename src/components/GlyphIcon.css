.GlyphIcon {
  width: 1.846em;
  height: 1.846em;
  border-radius: 100px;
  background-color: #E9ECEF;
  display: inline-block;
  background-repeat: no-repeat;
  background-position: center; 
  background-size: 10px 10px;
}

.GlyphIcon_ArrowLeft {
  background-image: url(./ArrowLeft.svg);
  background-position: 45% 50%;
}

.GlyphIcon_ArrowRight {
  background-image: url(./ArrowRight.svg);
  background-position: 50% 50%;
}