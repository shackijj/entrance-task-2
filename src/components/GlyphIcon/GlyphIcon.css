/* Generated by https://icomoon.io/ */

@font-face {
  font-family: 'GlyphIcon';
  src:  url(./GlyphIcon.eot);
  src:  url(./GlyphIcon.eot) format('embedded-opentype'),
    url(./GlyphIcon.ttf) format('truetype'),
    url(./GlyphIcon.woff) format('woff'),
    url(./GlyphIcon.svg) format('svg');
  font-weight: normal;
  font-style: normal;
}

.GlyphIcon {
  /* use !important to prevent issues with browser extensions that change fonts */
  font-family: 'GlyphIcon' !important;
  speak: none;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;

  /* Better Font Rendering =========== */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.GlyphIcon_ArrowLeft:before {
  content: "\e900";
}
.GlyphIcon_ArrowRight:before {
  content: "\e901";
}
.GlyphIcon_Calendar:before {
  content: "\e902";
}
.GlyphIcon_Close:before {
  content: "\e903";
}
.GlyphIcon_Edit:before {
  content: "\e904";
}