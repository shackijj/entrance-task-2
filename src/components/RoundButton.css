.RoundButton {
  display: inline-block;
  position: relative;
  width: 1.846em;
  height: 1.846em;
  border-radius: 100px;
  color: rgba(0,0,0,0.43);
  background-color: #E9ECEF;
  cursor: pointer;
}


.RoundButton-Icon {
  position: absolute;
  top: 55%;
  left: 50%;
  transform: translate(-50%, -50%);
}