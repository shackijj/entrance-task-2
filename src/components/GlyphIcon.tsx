import * as React from 'react';
import './GlyphIcon.css';
import * as classNames from 'classnames';

interface GlyphIconProps {
  classes?: string[];
}

const GlyphIcon = ({classes}: GlyphIconProps) => (
  <span className={classNames('GlyphIcon', classes)}/>
);

export default GlyphIcon;
