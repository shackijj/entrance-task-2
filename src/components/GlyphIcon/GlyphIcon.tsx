import * as React from 'react';
import './GlyphIcon.css';
import * as classNames from 'classnames';

interface IconProps {
  classes?: string[];
  onClick?: () => void;
}

export const ArrowLeft: React.SFC<IconProps> =
  ({classes, onClick}) =>
    (<span onClick={onClick} className={classNames(['GlyphIcon', 'GlyphIcon_ArrowLeft', classes])}/>);
export const ArrowRight: React.SFC<IconProps> =
  ({classes, onClick}) =>
    (<span onClick={onClick} className={classNames(['GlyphIcon', 'GlyphIcon_ArrowRight', classes])}/>);
export const Calendar: React.SFC<IconProps> = 
  ({classes, onClick}) =>
    (<span onClick={onClick} className={classNames(['GlyphIcon', 'GlyphIcon_Calendar', classes])}/>);
export const Close: React.SFC<IconProps> = 
  ({classes, onClick}) =>
    (<span onClick={onClick} className={classNames(['GlyphIcon', 'GlyphIcon_Close', classes])}/>);
export const Edit: React.SFC<IconProps> =
  ({classes, onClick}) =>
    (<span onClick={onClick} className={classNames(['GlyphIcon', 'GlyphIcon_Edit', classes])}/>);