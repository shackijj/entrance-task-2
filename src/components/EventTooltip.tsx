import * as React from 'react';
import User from './User';
import RoundButton from './RoundButton';
import { Edit } from './GlyphIcon/GlyphIcon';
import * as moment from 'moment';
import './EventTooltip.css';

import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

type User = {
  login: string;
  avatarUrl: string;
};

type Room = {
  title: string;
};

type Event = {
  title: string;
  dateStart: string;
  dateEnd: string;
  room: Room;
  users: User[];
};

interface EventTooltipProps extends Event {
  style?: React.CSSProperties;
}

export const EventTooltip: React.SFC<EventTooltipProps> = ({title, room, dateStart, dateEnd, users, style}) => {
  const momentStart = moment(dateStart);
  const momentEnd = moment(dateEnd);
  const date = momentStart.format('D MMMM');
  const start = momentStart.format('HH:mm');
  const end = momentEnd.format('HH:mm');
  
  let moreUsers;
  const moreUsersLen = users.length - 1;
  switch (moreUsersLen) {
    case 0:
      moreUsers = '';
      break;
    case 1:
      moreUsers = `и еще 1 участник`;
      break;
    case 2:
    case 3:
    case 4:
      moreUsers = `и еще ${moreUsersLen} участника`;
      break;
    default:
      moreUsers = `и еще ${moreUsersLen} участников`;
  } 

  return (
    <div className="EventTooltip" style={style}>
      <RoundButton classes={['EventTooltip-Button']} icon={<Edit/>}/>
      <div className="EventTooltip-Title">{title}</div>
      <div className="EventTooltip-Info">
        {date}, {start}—{end} · {room.title}
      </div>
      <div className="EventTooltip-Users">
        <User {...users[0]}/>
        <span className="EventTooltip-MoreUsers">{moreUsers}</span>
      </div>
    </div>
  );
};

const TOOLTIP_QUERY = gql`
  query EventQuery($id: ID!) {
    event(id: $id) {
      dateStart
      dateEnd
      room {
        title
      }
      users {
        login
        avatarUrl
      }
    }
  }
`;

type Response = {
  event: Event;
};

type InputProps = {
  id: string;
  style?: React.CSSProperties;
};

export const withGraphQL = graphql<Response, InputProps>(TOOLTIP_QUERY, {
  options: ({id}) => ({
    variables: { id }
  })
});

export default withGraphQL(({data, style}) => {
  if (data && data.event) {
    return <EventTooltip {...data.event} style={style}/>;
    
  }
  return <div/>;
});