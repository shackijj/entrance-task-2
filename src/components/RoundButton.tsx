import * as React from 'react';
import './RoundButton.css';
import * as classNames from 'classnames';

interface RoundButtonProps {
  icon: JSX.Element;
  classes?: string[];
  onClick?: () => void;
}

const RoundButton: React.SFC<RoundButtonProps> = ({classes, icon, onClick}) => (
  <a className={classNames('RoundButton', classes)} onClick={onClick}>
    <span className="RoundButton-Icon">
      {icon}
    </span>
  </a>
);

export default RoundButton;
