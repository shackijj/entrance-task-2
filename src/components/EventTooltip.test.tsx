import * as React from 'react';
import EventTooltip from './EventTooltip';
import User from './User';
import RoundButton from './RoundButton';
import { Edit } from './GlyphIcon/GlyphIcon';
import { shallow } from 'enzyme';

describe('EventTooltip', () => {
  it('should render an event', () => {
    const event = {
      title: 'Событие 3',
      dateStart: new Date('2018-01-09T18:30:00.55'),
      dateEnd: new Date('2018-01-09T20:45:00.55'),
      room: {
        title: 'Желтый дом'
      },
      users: [
        {
          name: 'Дарт Вейдер 1',
          avatarUrl: 'https://pp.userapi.com/c402317/v402317531/6107/Tg1PWZHDAO0.jpg'
        },
        {
          name: 'Дарт Вейдер 2',
          avatarUrl: 'https://pp.userapi.com/c402317/v402317531/6107/Tg1PWZHDAO0.jpg'
        },
        {
          name: 'Дарт Вейдер 3',
          avatarUrl: 'https://pp.userapi.com/c402317/v402317531/6107/Tg1PWZHDAO0.jpg'
        }
      ]
    };

    const actual = shallow(<EventTooltip {...event} />);
    const expected = (
      <div className="EventTooltip">
        <RoundButton classes={['EventTooltip-Button']} icon={<Edit/>}/>
        <div className="EventTooltip-Title">
          Событие 3
        </div>
        <div className="EventTooltip-Info">
          9 января, 18:30—20:45 · Желтый дом
        </div>
        <div className="EventTooltip-Users">
          <User {...event.users[0]}/>
          <span className="EventTooltip-MoreUsers">и еще 2 участника</span>
        </div>
      </div>
    );
    expect(actual.matchesElement(expected)).toBeTruthy();
  });
});