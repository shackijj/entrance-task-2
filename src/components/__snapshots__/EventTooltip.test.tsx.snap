// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventTooltip should render an event 1`] = `
<div
  className="EventTooltip"
>
  <RoundButton
    classes={
      Array [
        "EventTooltip-Button",
      ]
    }
    icon={<Unknown />}
  />
  <div
    className="EventTooltip-Title"
  >
    Событие 3
  </div>
  <div
    className="EventTooltip-Info"
  >
    9 January
    , 
    18:30
    —
    20:45
     · 
    Желтый дом
  </div>
  <div
    className="EventTooltip-Users"
  >
    <User
      avatarUrl="https://pp.userapi.com/c402317/v402317531/6107/Tg1PWZHDAO0.jpg"
      name="Дарт Вейдер 1"
    />
    <span
      className="EventTooltip-MoreUsers"
    >
      и еще 2 участника
    </span>
  </div>
</div>
`;
